Source/AliveLibAE/Renderer/OpenGLRenderer.cpp

```diff
@@ -0,0 +1,1519 @@
+        }
+    }
+    ImGui::End();
+    
+}
+
+void OpenGLRenderer::Destroy()
+{
+    ImGui_ImplSDL2_Shutdown();
+
+    mTextureShader.Free();
+
+    for (auto& t : gRendererTextures)
+    {
+        glDeleteTextures(1, &t.mTextureID);
+    }
+
+    for (auto& t : gRendererPals)
+    {
+        glDeleteTextures(1, &t.mPalTextureID);
+    }
+
+    glDeleteTextures(1, &gDecodedTextureCache);
+
+    if (mContext)
+    {
+        SDL_GL_DeleteContext(mContext);
+        mContext = nullptr;
+    }
+}
+
+bool OpenGLRenderer::Create(TWindowHandleType window)
+{
+    mWindow = window;
+    mWireframe = false;
+
+    // Find the opengl driver
+    const int numDrivers = SDL_GetNumRenderDrivers();
+    if (numDrivers < 0)
+    {
+        LOG_ERROR("Failed to get driver count " << SDL_GetError());
+    }
+
+    LOG_INFO("Got " << numDrivers << " drivers");
+
+    int index = -1;
+    for (int i = 0; i < numDrivers; i++)
+    {
+        SDL_RendererInfo info = {};
+        if (SDL_GetRenderDriverInfo(i, &info) < 0)
+        {
+            LOG_WARNING("Failed to get render " << i << " info " << SDL_GetError());
+        }
+        else
+        {
+            LOG_INFO(i << " name " << info.name);
+            if (strstr(info.name, "opengl"))
+            {
+                index = i;
+                break;
+            }
+        }
+    }
+
+    if (index == -1)
+    {
+        LOG_WARNING("OpenGL SDL2 driver not found");
+        return false;
+    }
+
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 3);
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 1);
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE);
+
+    // Create context
+    mContext = SDL_GL_CreateContext(window);
+    if (mContext == NULL)
+    {
+        LOG_ERROR("OpenGL context could not be created! SDL Error: " << SDL_GetError());
+        return false;
+    }
+    else
+    {
+        // Initialize GLEW
+        glewExperimental = GL_TRUE;
+        GLenum glewError = glewInit();
+        if (glewError != GLEW_OK)
+        {
+            LOG_ERROR("Error initializing GLEW! " << glewGetErrorString(glewError));
+        }
+
+        // Use Vsync
+        if (SDL_GL_SetSwapInterval(1) < 0)
+        {
+            LOG_ERROR("Warning: Unable to set VSync! SDL Error: " << SDL_GetError());
+        }
+    }
+
+    ImGui::CreateContext();
+
+    // Setup IMGUI for texture debugging
+    ImGui_ImplSDL2_InitForOpenGL(mWindow, mContext);
+    ImGui_ImplOpenGL3_Init("#version 150");
+
+    // Create our render buffers
+    glGenVertexArrays(1, &mVAO);
+    glBindVertexArray(mVAO);
+    glGenBuffers(1, &mIBO);
+    glGenBuffers(1, &mVBO);
+
+    // Set our Projection Matrix, so stuff doesn't get rendered in the quantum realm.
+    m_View = glm::ortho<float>(0, 640, 240, 0, 0, 1);
+
+    // TODO: Even worth implementing?
+    mVRamTexture = Renderer_CreateTexture();
+
+    mTextureShader.LoadFromFile("shaders/texture.vsh", "shaders/texture.fsh");
```

Comment:
it might be worth #including these to a static const char buffer, copying 1 exe and a dll is already too much for some people